### DISCUSSION TITLE
Plugins model error

### DISCUSSION THREAD

**Discussion by ZubikIT** [danny-avila/LibreChat#3291] [Troubleshooting]
### What happened?

Get error

### Steps to Reproduce

1. Goto .env and set OPENAI_API_KEY
2. docker compose up
3. Goto Plugins, select gpt-4
4. Select calculator
5. Write 2+4
6. Press enter
7. See error 
Something went wrong. Here's the specific error message we encountered: Encountered an error while attempting to respond: 404 functions is not supported in this model. For a list of supported models, refer to https://platform.openai.com/docs/guides/function-calling/supported-models.

If remove OPENAI_API_KEY from .env and insert to interface UI plugins all work.

### What browsers are you seeing the problem on?

Something went wrong. Here's the specific error message we encountered: Encountered an error while attempting to respond: 404 functions is not supported in this model. For a list of supported models, refer to https://platform.openai.com/docs/guides/function-calling/supported-models.

### Relevant log output
correct calculate 

<img width="1087" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/78b5da1e-f163-4abc-a757-92bc01aab559">

### Screenshots

<img width="1440" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/c596f595-b879-4480-93e0-b07b14eb40cb">
<img width="1440" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/292cd99e-046c-4da5-bf87-cddfaa88cb58">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ZubikIT** [comment]
I found where get default model "gpt-4-32k-0314"
  getFunctionModelName(input) in PluginsClient.js
<img width="860" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/3f0d1901-8bf7-47aa-86e4-330049db5941">
<img width="1439" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/421278e4-d5ef-4998-8693-05d8fc5b8e4f">

I'm right?

**Comment by danny-avila** [comment]
Select the agent model in settings

![image](https://github.com/danny-avila/LibreChat/assets/110412045/c1fdc83e-ce86-42c5-b2a5-04267f61c4fb)
![Screenshot 2024-07-08 081056](https://github.com/danny-avila/LibreChat/assets/110412045/dc3c40ba-5562-41e8-bb4b-ad6338547d39)

**Reply by danny-avila** [reply]
Also I’d like to add that Plugins are now deprecated, and you should use Agents instead:
https://librechat.ai/docs/features/agents

**Comment by ZubikIT** [comment]
Thanks very match work!
<img width="1437" alt="image" src="https://github.com/danny-avila/LibreChat/assets/745561/289fa23b-97e9-4bfd-b133-1fef2ca0529c">

**Comment by santhoshsram** [comment]
@danny-avila Thanks. I had the same issue and resolved it by following your steps. From the error it was difficult to figure out what the issue was. Also, the default model set to "chatgpt-4o-latest" was causing the issue. Should the default be switched to gpt-4o? gpt-4o-latest is also cheaper than chatgpt-4o-latest.

**Reply by danny-avila** [reply]
There is no default, the first model is used that is defined in your models list. Plugins will be deprecated soon, anyway

**Reply by urza-lab** [reply]
will there be embedded functionality for browsing, or do you expect openAI integrating the function into the API?
